Update pypi.python.org links to pypi.org links

pgadige opened the pull request:
Fixes #476 

Updated `pypi.python.org` links to `pypi.org` links.
Changed `https://pypi.python.org/pypi?%3Aaction=list_classifiers` to `https://pypi.org/classifiers/`

di commented on source/guides/migrating-to-pypi-org.rst:
```diff
@@ -40,7 +40,7 @@ configured in a file located at ``$HOME/.pypirc``. If you see a file like:
 
     [pypi]
-    repository:https://pypi.python.org/pypi
+    repository:https://pypi.org/pypi
```
This should not change, as it represents the state of `.pypirc` _before_ migrating, so it may have old links.

pgadige replied:
Thank you for the feedback. I've reverted to old link.

pgadige commented:
Thank you for the feedback, @di! 
I changed links of the form `https://pypi.org/pypi/something` to `https://pypi.org/project/something` to avoid redirects.
I also changed `https://pypi.python.org/pypi?%3Aaction=list_classifiers` to `https://pypi.org/classifiers/` in one more file, `source/specifications/core-metadata.rst`

jonparrott commented:
Thank you, @pgadige and @di!
